src/Functional.CQS.AOP.IoC.PureDI.Caching/Models/DataWrapper.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Functional.CQS.AOP.IoC.PureDI.Caching.Models
+{
+	/// <summary>
+	/// Encapsulates a reference-type value.
+	/// </summary>
+	/// <typeparam name="T">The type.</typeparam>
+	/// <remarks>
+	/// Used to ensure that cache implementations never store NULL, and to ensure that all types assignable to <typeparamref name="T"/> are treated as <typeparamref name="T"/>.
```

Comment:
The latter half of the sentence may be confusing...  It's _supposed_ to imply that if you're expecting a query to return `IFoo` and you have classes `Bar`, `Baz` implementing `IFoo`, you can store `Bar`, `Baz` types in the cache and retrieval will resolve them as `IFoo`.

Reply:
I don't really see how to re-word that any better, so I think this is good.